lib/class-wp-rest-url-details-controller.php

```diff
@@ -228,21 +228,31 @@ private function get_title( $html ) {
 	/**
 	 * Parses the site icon from the provided HTML
 	 *
-	 * @param string $html the HTML from the remote website at URL.
-	 * @param string $url the target website URL.
-	 * @return string the icon URI (maybe empty).
+	 * @param string $html The HTML from the remote website at URL.
+	 * @param string $url  The target website URL.
+	 * @return string The icon URI (maybe empty).
 	 */
 	private function get_icon( $html, $url ) {
-		preg_match( '|<link.*?rel="\s*[shortcut]+(?:\s+[icon]+)*\s*".*?href="(.*?)".*?\/?>|is', $html, $matches );
```

Comment:
This pattern was causing a few problems:

- The order of attributes was fixed, meaning `rel` had to come before `href`
- Did not allow for multiline attributes
- Was picking multiple link elements when the first one didn't match the order and/or structure

This new implementation does the following:

- stays in the link element
- is not dependent on the order of attributes
- other attributes such as `type`, `size`, or custom ones can co-exist
- attributes can be on multiple lines